openfeature/openfeature.go

```diff
@@ -20,13 +20,23 @@ func initSingleton() {
 // SetProvider sets the default provider. Provider initialization is asynchronous and status can be checked from
 // provider status
 func SetProvider(provider FeatureProvider) error {
-	return api.setProvider(provider)
+	return api.setProvider(provider, true)
+}
+
+// SetProviderAndWait sets the default provider and waits for its initialization.
+func SetProviderAndWait(provider FeatureProvider) error {
+	return api.setProvider(provider, false)
```

Comment:
This function needs to return an error if init fails (and I think you should add documentation saying so)

https://openfeature.dev/specification/sections/flag-evaluation/#requirement-1124

Reply:
Good catch, I will improve this

Reply:
Done - https://github.com/open-feature/go-sdk/pull/251/commits/54ef5b0457b26309d3d4215eac8abc148de1a615